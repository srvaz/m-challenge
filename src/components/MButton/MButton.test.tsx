import { BUTTON_VARIANTS, MButton } from './index';
import { fireEvent, render } from '@testing-library/react';

import React from 'react';

describe('MButton', () => {
  let wrapper: any;
  const props = {
    onClick: jest.fn(),
    variant: BUTTON_VARIANTS.DEFAULT,
  }

  describe('should render correctly', () => {
    test('default', () => {
      wrapper = render(<MButton {...props}>Mock Button</MButton>);
      expect(wrapper.asFragment()).toMatchSnapshot();
    });

    test('primary', () => {
      props.variant = BUTTON_VARIANTS.PRIMARY;
      wrapper = render(<MButton {...props}>Mock Button</MButton>);
      expect(wrapper.asFragment()).toMatchSnapshot();
    });

    test('secondary', () => {
      props.variant = BUTTON_VARIANTS.SECONDARY;
      wrapper = render(<MButton {...props}>Mock Button</MButton>);
      expect(wrapper.asFragment()).toMatchSnapshot();
    });

    test('outlined', () => {
      props.variant = BUTTON_VARIANTS.OUTLINED;
      wrapper = render(<MButton {...props}>Mock Button</MButton>);
      expect(wrapper.asFragment()).toMatchSnapshot();
    });
  })

  test('should execute the callback function', () => {
    wrapper = render(<MButton {...props}>Mock Button</MButton>);
    fireEvent.click(wrapper.getByText(/Mock/));

    expect(props.onClick).toBeCalled();
  });
});

