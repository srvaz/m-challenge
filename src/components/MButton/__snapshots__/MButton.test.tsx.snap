// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MButton should render correctly default 1`] = `
<DocumentFragment>
  <button
    class="m-button m-button--default"
  >
    Mock Button
  </button>
</DocumentFragment>
`;

exports[`MButton should render correctly outlined 1`] = `
<DocumentFragment>
  <button
    class="m-button m-button--outlined"
  >
    Mock Button
  </button>
</DocumentFragment>
`;

exports[`MButton should render correctly primary 1`] = `
<DocumentFragment>
  <button
    class="m-button m-button--primary"
  >
    Mock Button
  </button>
</DocumentFragment>
`;

exports[`MButton should render correctly secondary 1`] = `
<DocumentFragment>
  <button
    class="m-button m-button--secondary"
  >
    Mock Button
  </button>
</DocumentFragment>
`;
